File: tournaments/arrayPreviousLess/arrayPreviousLess.cs
int[] arrayPreviousLess(int[] items) {
    var n = items.Length;
    var store = new Stack<int>();

    for (int i = 0; n > i; ++i) {
        var temp = items[i];
        while (0 < store.Count) {
            if (store.Peek() < items[i]) {
                items[i] = store.Peek();
                break;
            } else {
                store.Pop();
            }
        }

        if (0 == store.Count) {
            items[i] = -1;
        }

        store.Push(temp);
    }

    return items;
}

File: tournaments/arrayPreviousLess/arrayPreviousLess.cpp
std::vector<int> arrayPreviousLess(std::vector<int> items) {

  std::vector<int> result;
  for (int i = 0; i < items.size(); i++) {
    int substitute = -1;
    for (int j = 0; j < i; j++) {
      if (items[j] < items[i]) {
        substitute = items[j];
      }
    }
    result.push_back(substitute);
  }

  return result;
}
